src/server/pps/server/api_server.go

```diff
@@ -1827,8 +1827,8 @@ func (a *apiServer) validatePipelineRequest(request *pps.CreatePipelineRequest)
 	return nil
 }
 
-func (a *apiServer) validateEnterpriseChecks(ctx context.Context, req *pps.CreatePipelineRequest) error {
-	if _, err := a.inspectPipeline(ctx, req.Pipeline, false); err == nil {
+func (a *apiServer) validateEnterpriseChecksInTransaction(ctx context.Context, txnCtx *txncontext.TransactionContext, req *pps.CreatePipelineRequest) error {
```

Comment:
nit: I think you can keep the name of the function validateEnterpriseChecks since there aren't two versions of the function